json/src/spec/params.rs

```diff
@@ -149,4 +150,21 @@ mod tests {
 		assert_eq!(deserialized.gas_limit_bound_divisor, Uint(U256::from(0x20)));
 		assert_eq!(deserialized.max_code_size, Some(Uint(U256::from(0x1000))));
 	}
+
+	#[test]
+	#[should_panic(expected = "a non-zero value")]
+	fn test_zero_value_divisor() {
+		let s = r#"{
+			"maximumExtraDataSize": "0x20",
+			"networkID" : "0x1",
+			"chainID" : "0x15",
+			"subprotocolName" : "exp",
+			"minGasLimit": "0x1388",
+			"accountStartNonce": "0x01",
+			"gasLimitBoundDivisor": "0x0",
+			"maxCodeSize": "0x1000"
+		}"#;
+
+		let _deserialized: Params = serde_json::from_str(s).unwrap();
```

Comment:
how does printed error look like? Does it say describe which field is expected to have `a non-zero value`?

Maybe instead of using `should_panic` you could do:

```rust
assert_eq!(serde_json::from_str(s).unwrap_err().to_string(), "error message");
```

Reply:
It doesn't specify which field is failing, only it's position. The error message looks like this:
```"invalid value: integer `0`, expected a non-zero value at line 3 column 39"```. I agree the test can be converted not to use `should_panic`. It would be nice to show the field name but this is the same behavior we already have for other fields.

Reply:
Why is it better to avoid `#[should_panic(expect)]`? I thought that using an attribute would make more sense, since it's available.